### PR TITLE
Correct regex pattern for BCP-47 language tags

### PR DESCRIPTION
**Pull request by vcschapp**
# Description

*Brief description of the business purpose and effect of the pull request.*

This PR has two changes in it:

1. Correct regex pattern for BCP-47 language tags. This fixes "false negative" validation failures on valid BCP-47 language tags like *zh-Latn-pinyin*, and is a partial solution to https://github.com/OvertureMaps/schema-wg/issues/293.
2. Fix bug in test script where special characters broke error matching.

# Reference

*List of relevant links to GitHub issues, PRs, and other documentation.*

1. *schema-wg* issue 293: [Open
[[Bug]: Many features fail names.common language validation across all themes](https://github.com/OvertureMaps/schema-wg/issues/293#top)](https://github.com/OvertureMaps/schema-wg/issues/293)

# Testing

*Brief description of the testing done for this change showing why you are confident it works as expected and does not introduce regressions. Provide sample output data where appropriate.* 

- Updated counterexamples.
- Added a new counterexample.
- Added a new example.

# Checklist

*Checklist of tasks commonly-associated with schema pull requests. Please review the relevant checklists and ensure you do all the tasks that are required for the change you made.*

1. [x] Add relevant examples.
3. [x] Add relevant counterexamples.
4. [ ] Update in-schema documentation using plain English written in complete sentences, if an update is required.
5. [ ] Update Docusaurus documentation, if an update is required.
6. [ ] Review change with Overture technical writer to ensure any advanced documentation needs will be taken care of, unless the change is trivial and would not affect the documentation.

# Documentation Website

*Update the hyperlink below to put the pull request number in.*

[Docs preview for this PR.](https://dfhx9f55j8eg5.cloudfront.net/pr/165)

### CODE REVIEW

File: schema/defs.yaml [hunk 1]
```diff
@@ -179,7 +179,7 @@ description: Common schema definitions shared by all themes
         The complete lists of allowed values can be found at:
         https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry
       type: string
-      pattern: "^[a-zA-Z]{2,3}(-[a-zA-Z]{4})?(-[a-zA-Z]{2})?$"
+      pattern: "^(?:(?:[A-Za-z]{2,3}(?:-[A-Za-z]{3}){0,3}?)|(?:[A-Za-z]{4,8}))(?:-[A-Za-z]{4})?(?:-[A-Za-z]{2}|[0-9]{3})?(?:-(?:[A-Za-z0-9]{5,8}|[0-9][A-Za-z0-9]{3}))?(?:-[A-WY-Za-wy-z0-9](?:-[A-Za-z0-9]{2,8})+)?$"
```
Comment (jwass) [inline schema/defs.yaml:182]: Can you add a description of this regex?

**Review by jwass — APPROVED** [review]
Looks good. Thanks!

Would be nice if you could provide a description of that regex since they can be pretty difficult to parse visually. So if I have a tag that fails, it'll be tough to figure out why.

File: schema/defs.yaml (hunk 1 above)
Comment (vcschapp) [reply schema/defs.yaml:182]: Yup, next time I have a minute I'll update it.

Comment (vcschapp) [reply schema/defs.yaml]: Updated with documentation. Could you take a look, and resolve your comment if you feel it's addressed?

**Review by jenningsanderson — APPROVED** [review]
LGTM!

**Review by jwass — APPROVED** [review]

**Review by TristanDiet-TomTom — APPROVED** [review]

**Pull request merged**